src/components/timersDisplay/TabataDisplay.tsx

```diff
@@ -0,0 +1,103 @@
+import { useContext, useEffect, useState } from 'react';
+import { TimerContext } from '../contexts/context';
+import TimerDisplay from '../generic/TimerDisplay';
+
+interface TimerProps {
+    work: number;
+    rest: number;
+    rounds: number;
+}
+
+const Tabata = ({ work, rest, rounds }: TimerProps) => {
+    const globalTimerData = useContext(TimerContext);
+    const [seconds, setSeconds] = useState(-1);
+    const [roundsRemaining, setRoundsRemaining] = useState(rounds);
+    const [isWorking, setIsWorking] = useState(true);
+    const [cacheChecked, setCacheChecked] = useState(false);
+
+    useEffect(() => {
+        if (globalTimerData.hardReset) {
+            setSeconds(work);
+            setRoundsRemaining(rounds - 1);
+            setIsWorking(true);
+        }
+        if (globalTimerData.timerComplete) {
+            setSeconds(work);
+            setRoundsRemaining(rounds - 1);
+            setIsWorking(true);
+        }
+        if (globalTimerData.newTimer) {
+            setSeconds(work);
+            setRoundsRemaining(rounds - 1);
+            setIsWorking(true);
+            globalTimerData.setNewTimer(false);
+        }
+        if (seconds === -1) {
+            const lS = Number(localStorage.getItem('seconds'));
+            if (lS === -1) {
+                setSeconds(work);
+            } else {
+                setSeconds(Number(localStorage.getItem('seconds')));
+            }
+        }
+        if (localStorage.getItem('roundsRemaining') !== '-1' && !globalTimerData.hardReset) {
+            setRoundsRemaining(Number(localStorage.getItem('roundsRemaining')));
+        }
+        if (localStorage.getItem('isWorking') !== '-1' && !globalTimerData.hardReset) {
+            if (localStorage.getItem('isWorking') === 'false') setIsWorking(false);
+            else setIsWorking(true);
+        }
+        setCacheChecked(true);
+    }, [globalTimerData, seconds, work, rounds]);
+
+    useEffect(() => {
+        if (cacheChecked) {
+            let interval = null;
+            localStorage.setItem('seconds', seconds.toString());
```

Comment:
use custom hooks to abstract this and avoid repeating. we covered this in lecture